client/app/api/course/Materials.js

```diff
@@ -0,0 +1,20 @@
+import BaseCourseAPI from './Base';
+
+export default class MaterialsAPI extends BaseCourseAPI {
+  /**
+  * Destroy the material.
+  *
+  * @param {number} folderId
+  * @param {numbrt} materialId
+  * @return {Promise}
+  * success response: {}
+  * error response: { :message }
+  */
+  destroy(folderId, materialId) {
+    return this.getClient().post(`${this._getUrlPrefix()}/${folderId}/files/${materialId}`);
+  }
+
+  _getUrlPrefix() {
```

Comment:
`_` is used for private methods, it's just a convention to let you know the method is private, it doesn't prevent you from actually calling it. Any concerns if we follow this rule? 

Can see some discussion here:
http://stackoverflow.com/questions/27849064/how-to-implement-private-method-in-es6-class-with-traceur
http://eslint.org/docs/rules/no-underscore-dangle